File: app/src/main/java/com/veyndan/paper/reddit/deeplink/CustomDeepLinkHandler.java
package com.veyndan.paper.reddit.deeplink;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.airbnb.deeplinkdispatch.DeepLinkHandler;

@DeepLinkHandler(AppDeepLinkModule.class)
public class CustomDeepLinkHandler extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        final DeepLinkDelegate deepLinkDelegate = new DeepLinkDelegate(new AppDeepLinkModuleLoader());

        deepLinkDelegate.dispatchFrom(this);

        finish();
    }
}

File: app/src/main/java/com/veyndan/paper/reddit/deeplink/CustomDeepLinkHandler.kt
package com.veyndan.paper.reddit.deeplink

import android.os.Bundle
import android.support.v7.app.AppCompatActivity

import com.airbnb.deeplinkdispatch.DeepLinkHandler

@DeepLinkHandler(AppDeepLinkModule::class)
class CustomDeepLinkHandler : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val deepLinkDelegate = DeepLinkDelegate(AppDeepLinkModuleLoader())

        deepLinkDelegate.dispatchFrom(this)

        finish()
    }
}
